### PR TITLE
core-*-api: backport support for external route ref default targets + config route binding

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

The new frontend system supports route binding through static configuration, for example:

```yaml
app:
  routes:
    bindings:
      catalog.createComponent: catalog-import.importPage
```

It also lets you define default targets for external route references, for example:

```ts
export const registerComponentRouteRef = createExternalRouteRef({
  optional: true,
  defaultTarget: 'catalog-import.importPage',
});
```

This adds support for both of these features in the existing frontend system.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — COMMENTED** [review]
nice

**Review by acierto — APPROVED** [review]

File: packages/core-app-api/src/app/AppManager.tsx [hunk 1]
```diff
@@ -307,6 +297,24 @@ DEPRECATION WARNING: React Router Beta is deprecated and support for it will be
...
+        );
+
+        try {
+          validateRouteParameters(routing.paths, routing.parents);
+          validateRouteBindings(routeBindings, this.plugins);
+        } catch (error) {
+          routeValidationError = error;
+          throw error;
+        }
+      } else if (routeValidationError) {
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/core-app-api/src/app/AppManager.tsx:314]: Ah yep good call, I've moved the resolution into the same block as the validation

File: packages/core-compat-api/src/convertLegacyRouteRef.ts [hunk 2]
```diff
@@ -199,10 +203,8 @@ export function convertLegacyRouteRef(
       getDescription() {
         return legacyRefStr;
       },
-      getDefaultTarget() {
-        // TODO(freben): These are not yet supported in the old system; just returning undefined for now
-        return undefined;
-      },
+      // This might already be implemented in the legacy ref, but we override it just to be sure
+      getDefaultTarget: newRef.getDefaultTarget,
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/core-compat-api/src/convertLegacyRouteRef.ts:207]: aight, made it a bit safer :p

File: packages/core-app-api/src/app/AppManager.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (freben) [reply packages/core-app-api/src/app/AppManager.tsx:314]: Still wanna move this to _above_ `if (!routeBindings)` right?

**Review by freben — APPROVED** [review]

**Pull request merged**